### ISSUE TITLE
Composite keys join

### ISSUE THREAD

**Issue by hogggerty** [staudenmeir/eloquent-has-many-deep#23]
I'm not sure, that is that package functionality, but it would be great, if you add ability to use composite keys on creating relations. 

As example:
```SQL
SELECT
    a.* 
FROM
    a_table AS a
INNER JOIN
    b_table AS b
ON 
    a.name = b.name
    AND
    a.surname = b.surname
```

**Comment by staudenmeir** [comment]
Do you mean for a `HasManyDeep` relationship or for one of Laravel's regular relationships?

**Comment by hogggerty** [comment]
For HasManyDeep relationship.

**Comment by staudenmeir** [comment]
You can use this package for the regular relationships: https://github.com/topclaudy/compoships

I'm not sure whether it's possible to add composite keys to `HasManyDeep`. I'll take a look.

**Comment by staudenmeir** [comment]
Can you provide your `HasManyDeep` relationship that requires composite keys?

**Comment by staudenmeir** [comment]
I released `v1.16` (Laravel 9) with support for composite keys.

If you concatenate existing relationships with `hasManyDeepFromRelations()`, you can now include `compoships` relationships:
https://github.com/staudenmeir/eloquent-has-many-deep#third-party-packages

If you define relationships manually with `hasManyDeep()`, you can now use the `CompositeKey` class:
https://github.com/staudenmeir/eloquent-has-many-deep#composite-keys
